### PR TITLE
std.file unittest doc example

### PR DESCRIPTION
**Pull request by burner**
move some examples from comments to unittest

### CODE REVIEW

File: std/file.d [hunk 1]
```diff
@@ -245,6 +233,17 @@ void[] read(R)(R name, size_t upTo = size_t.max)
         return readImpl(null, name.tempCString!FSChar(), upTo);
 }
 
+///
+@safe unittest
+{
+    import std.utf : byChar;
+    scope(exit) { assert(exists(deleteme)); remove(deleteme); }
```
Comment (quickfur) [inline std/file.d]: I'm not sure `deleteme` is a good name to use as an example in the documentation.

Comment (burner) [inline std/file.d]: changed

File: std/file.d [hunk 2]
```diff
@@ -452,6 +427,16 @@ void write(R)(R name, const void[] buffer)
         writeImpl(null, name.tempCString!FSChar(), buffer, false);
 }
 
+///
+unittest
+{
+   scope(exit) { assert(exists(deleteme)); remove(deleteme); }
```
Comment (quickfur) [inline std/file.d]: Probably should rename `deleteme` here too?

File: std/file.d [hunk 3]
```diff
@@ -484,6 +455,18 @@ void append(R)(R name, const void[] buffer)
         writeImpl(null, name.tempCString!FSChar(), buffer, true);
 }
 
+///
+unittest
+{
+   scope(exit) { assert(exists(deleteme)); remove(deleteme); }
```
Comment (quickfur) [inline std/file.d]: ditto

File: std/file.d [hunk 4]
```diff
@@ -3559,12 +3535,15 @@ slurp(Types...)(string filename, in char[] format)
...
 
+///
 unittest
 {
-    // Tuple!(int, double)[] x;
-    // auto app = appender(&x);
-    write(deleteme, "12 12.25\n345 1.125");
     scope(exit) { assert(exists(deleteme)); remove(deleteme); }
+
+    write(deleteme, "12 12.25\n345 1.125");
```
Comment (quickfur) [inline std/file.d]: ditto

File: std/file.d [hunk 5]
```diff
@@ -215,18 +215,6 @@ Read entire contents of file $(D name) and returns it as an untyped
 array. If the file size is larger than $(D upTo), only $(D upTo)
 bytes are read.
 
-Example:
-
-----
-import std.file, std.stdio;
-void main()
-{
-   auto bytes = cast(ubyte[]) read("filename", 5);
```
Comment (DmitryOlshansky) [inline std/file.d]: Where is DDoc-ed version? Removing it just because it can't be tested as is not good.

Comment (burner) [inline std/file.d]: fixed

**Comment by burner** [comment]
@DmitryOlshansky @quickfur fixed

**Comment by DmitryOlshansky** [comment]
LGTM

File: std/file.d [hunk 6]
```diff
@@ -484,6 +464,22 @@ void append(R)(R name, const void[] buffer)
...
+   {
+       assert(exists("someUnqiueFilename"));
+       remove("someUnqiueFilename");
+   }
+
+   int[] a = [ 0, 1, 1, 2, 3, 5, 8 ];
+   write("someUnqiueFilename", a);
+   int[] b = [ 13, 21 ];
+   append("someUnqiueFilename", b);
+   assert(cast(int[]) read("someUnqiueFilename") == a ~ b);
```
Comment (quickfur) [inline std/file.d]: The filename is certainly unique, but it looks like a typo. :-)  `someUniqueFilename`, not `someUnqiueFilename`.

Comment (burner) [inline std/file.d]: ;-)

**Comment by quickfur** [comment]
LGTM

**Comment by quickfur** [comment]
Auto-merge toggled on

**Pull request merged**